Jetsnack/app/src/main/java/com/example/jetsnack/ui/utils/SystemUi.kt

```diff
@@ -19,12 +19,90 @@ package com.example.jetsnack.ui.utils
 import android.os.Build
 import android.view.View
 import android.view.Window
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.Providers
+import androidx.compose.runtime.remember
 import androidx.compose.runtime.staticAmbientOf
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.graphics.compositeOver
 import androidx.compose.ui.graphics.luminance
 import androidx.compose.ui.graphics.toArgb
 
+private val AmbientStatusBar = staticAmbientOf { UnspecifiedSystemBar }
+
+private val AmbientNavigationBar = staticAmbientOf { UnspecifiedSystemBar }
+
+@Composable
+fun StatusBar(
+    color: Color,
+    darkIcons: Boolean = color.luminance() > 0.5f,
+    transformColorForLightContent: (Color) -> Color = BlackScrimmed,
+    content: @Composable () -> Unit
+) {
+    val statusBar = SystemBar(color, darkIcons, transformColorForLightContent)
+    val controller = AmbientSysUiController.current
+    val status = AmbientStatusBar.current
+    val prevStatus = remember { status }
+    DisposableEffect(statusBar, prevStatus) {
+        controller.setStatusBarColor(color, darkIcons, transformColorForLightContent)
+        onDispose {
+            controller.setStatusBarColor(
+                prevStatus.color,
+                prevStatus.darkIcons,
+                prevStatus.transformColorForLightContent
+            )
+        }
+    }
+    Providers(AmbientStatusBar provides statusBar, content = content)
+}
+
+@Composable
+fun NavigationBar(
+    color: Color,
+    darkIcons: Boolean = color.luminance() > 0.5f,
+    transformColorForLightContent: (Color) -> Color = BlackScrimmed,
+    content: @Composable () -> Unit
+) {
+    val navBar = SystemBar(color, darkIcons, transformColorForLightContent)
+    val controller = AmbientSysUiController.current
+    val nav = AmbientNavigationBar.current
+    val prevNav = remember { nav }
+    DisposableEffect(navBar, prevNav) {
+        controller.setNavigationBarColor(color, darkIcons, transformColorForLightContent)
+        onDispose {
+            controller.setStatusBarColor(
+                prevNav.color,
+                prevNav.darkIcons,
+                prevNav.transformColorForLightContent
+            )
+        }
+    }
+    Providers(AmbientNavigationBar provides navBar, content = content)
+}
+
+@Composable
+fun SystemBars(
+    color: Color,
+    darkIcons: Boolean = color.luminance() > 0.5f,
+    transformColorForLightContent: (Color) -> Color = BlackScrimmed,
+    content: @Composable () -> Unit
+) {
+    StatusBar(
```

Comment:
We can probably be more optimal than nesting the calls.

Reply:
Maybe slightly but I don't think the increased code complexity is worth it. Feel strongly about this?

Reply:
Not particularly.

Reply:
👍